### PR TITLE
Commit a navigator.pop as soon as the back swipe is lifted

### LINKED ISSUE #29596: RouteObserver/NavigatorObserver.didPop no longer called on iOS back swipe
Hello, on master, deb, beta channel didPop no calls, it is bad, because need check screen pop,
```
final RouteObserver<PageRoute> routeObserver = RouteObserver<PageRoute>();

class _BaseWidgetState extends State<BaseWidget>
    with RouteAware, WidgetsBindingObserver {

@override
  void initState() {
    super.initState();
    WidgetsBinding.instance.addObserver(this);
  }

@override
  void didChangeDependencies() {
    super.didChangeDependencies();
    routeObserver.subscribe(this, ModalRoute.of(context));
  }

  @override
  void didPushNext() {
    super.didPushNext();
    print('didPushNext');
    if (widget.pushNext != null) widget.pushNext();
  }

  @override
  void didPop() {
    super.didPop();
    print('didPop');
    if (widget.pop != null) widget.pop();
  }

  @override
  void didPopNext() {
    super.didPopNext();
    print('didPopNext');
    if (widget.popNext != null) widget.popNext();
  }

  @override
  void didPush() {
    super.didPush();
    print('didPush');
    if (widget.push != null) widget.push();
  }

  @override
  void dispose() {
    routeObserver.unsubscribe(this);
    WidgetsBinding.instance.removeObserver(this);
    super.dispose();
  }

}
```

I test channel
1. stable
//push next screen
flutter: didPushNext
flutter: didPush
-----swipe screen(iOS)
       flutter: didPopNext
       flutter: didPop
-----touch back button (Android)
       flutter: didPopNext
       flutter: didPop

test: successful

2. master
//push next screen
flutter: didPushNext
flutter: didPush
-----swipe screen(iOS)
     no reaction!!!
-----touch back button (Android)
       flutter: didPopNext
       flutter: didPop

3. dev
//push next screen
flutter: didPushNext
flutter: didPush
----- swipe screen(iOS)
           no reaction!!!
-----touch back button (Android)
           flutter: didPopNext
           flutter: didPop

4. beta
//push next screen
flutter: didPushNext
flutter: didPush
-----swipe screen(iOS)
no reaction!!!
-----touch back button (Android)
flutter: didPopNext
flutter: didPop

Flutter 1.3.12-pre.29 • channel master • https://github.com/flutter/flutter.git
Framework • revision af6e09f6ea (34 hours ago) • 2019-03-18 21:40:21 -0700
Engine • revision 5088735e5f
Tools • Dart 2.2.1 (build 2.2.1-dev.1.0 None)

Running flutter doctor...
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, v1.3.14-pre.7, on Mac OS X 10.14.3 18D109, locale ru-RU)
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] iOS toolchain - develop for iOS devices (Xcode 10.1)
[✓] Android Studio (version 3.3)
[✓] Connected device (1 available)

• No issues found!

### PR DESCRIPTION
**Pull request by xster**
## Description

Not committing the pop as soon as it's irreversible makes it ambiguous what to signal to NavigatorObservers since other things can happen to the navigation stack in the meantime.

## Related Issues

Fixes #29596 

## Tests

I added the following tests:

*Replace this with a list of the tests that you added as part of this PR. A change in behaviour with no test covering it
will likely get reverted accidentally sooner or later. PRs must include tests for all changed/updated/fixed behaviors. See [Test Coverage].*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by xster** [comment]
@goderbauer can you review? Hans is out.

File: packages/flutter/test/cupertino/route_test.dart [hunk 1]
```diff
@@ -624,4 +638,41 @@ void main() {
     expect(tester.getTopLeft(find.text('1')).dx, moreOrLessEquals(0));
     expect(navigatorKey.currentState.userGestureInProgress, false);
   });
+
+  testWidgets('test edge swipe then drop back at ending point works', (WidgetTester tester) async {
+    await tester.pumpWidget(
```
Comment (goderbauer) [inline packages/flutter/test/cupertino/route_test.dart]: Add comment `/// Regression test for https://github.com/flutter/flutter/issues/29596.`?

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/test/cupertino/route_test.dart (hunk 1 above)
Comment (xster) [reply packages/flutter/test/cupertino/route_test.dart]: sg

**Comment by xster** [comment]
Thanks for reviewing

**Comment by 1AlexFix1** [comment]
Hooray

> Thanks for reviewing

**Pull request merged**